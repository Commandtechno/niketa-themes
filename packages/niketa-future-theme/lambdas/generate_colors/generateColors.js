import { piped, uniq } from 'rambdax'

import { getGradientBee } from '../../src/bees/getGradient'

export function generateColors({ input, levels = 20 }){
  const [ first, second ] = input

  const colors = piped(getGradientBee(first, second, levels), uniq)
  expect(colors).toMatchSnapshot()
}
