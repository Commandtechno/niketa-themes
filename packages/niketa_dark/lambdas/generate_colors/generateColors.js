import { flatten, piped, uniq } from 'rambdax'

import { translate } from '../../src/ants/mini/translate'
import { writeJsonAnt } from '../../src/ants/writeJson'
import { getGradientBee } from '../../src/bees/getGradient'
const base = 'lambdas/generate_colors/colors'

export function generateColors({ input, levels = 20, label = '' }){
  const [ firstRaw, secondRaw ] = input
  const OUTPUT = `${ base }/${ label }_COLORS.json`

  const first = firstRaw.startsWith('#') ? firstRaw : translate(firstRaw)

  const second = secondRaw.startsWith('#') ? secondRaw : translate(secondRaw)

  const colors = piped(getGradientBee(
    first, second, levels
  ), uniq)
  // log({ OUTPUT })

  return writeJsonAnt(OUTPUT, flatten(colors))
}
