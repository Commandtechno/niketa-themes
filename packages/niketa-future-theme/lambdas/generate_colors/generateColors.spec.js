import { generateColors } from './generateColors'

test('happy', () => {
  const input = [ '#b7bcbf', '#e9e9e9' ]

  expect(() =>
    generateColors({
      input,
      levels : 60,
    })).not.toThrow()
})
