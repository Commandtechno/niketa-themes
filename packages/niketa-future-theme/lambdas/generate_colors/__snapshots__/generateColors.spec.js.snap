// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Array [
  "#b7bcbf",
  "#b8bdc0",
  "#b9bec0",
  "#babec1",
  "#babfc2",
  "#bbc0c3",
  "#bcc1c3",
  "#bdc1c4",
  "#bec2c5",
  "#bfc3c5",
  "#bfc4c6",
  "#c0c4c7",
  "#c1c5c8",
  "#c2c6c8",
  "#c3c7c9",
  "#c4c7ca",
  "#c5c8ca",
  "#c5c9cb",
  "#c6cacc",
  "#c7cacd",
  "#c8cbcd",
  "#c9ccce",
  "#cacdcf",
  "#cacecf",
  "#cbced0",
  "#cccfd1",
  "#cdd0d2",
  "#ced1d2",
  "#cfd1d3",
  "#d0d2d4",
  "#d0d3d4",
  "#d1d4d5",
  "#d2d4d6",
  "#d3d5d6",
  "#d4d6d7",
  "#d5d7d8",
  "#d6d7d9",
  "#d6d8d9",
  "#d7d9da",
  "#d8dadb",
  "#d9dbdb",
  "#dadbdc",
  "#dbdcdd",
  "#dbddde",
  "#dcdede",
  "#dddedf",
  "#dedfe0",
  "#dfe0e0",
  "#e0e1e1",
  "#e1e1e2",
  "#e1e2e3",
  "#e2e3e3",
  "#e3e4e4",
  "#e4e4e5",
  "#e5e5e5",
  "#e6e6e6",
  "#e6e7e7",
  "#e7e7e8",
  "#e8e8e8",
  "#e9e9e9",
]
`;
